cpp/include/typedb/connection/DatabaseManager.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#pragma once
+
+#include <string>
+
+#include "typedb/common/native.hpp"
```

Comment:
The native header should _only_ be used in implementation files to avoid leaking it into the larger namespace. Where it is used as a field, such as here, you can use a pimpl pattern:
```cpp
// *.hpp
class DatabaseManager {
    // ...
    private:
        struct Impl;
        std::unique_ptr<Impl> pImpl;  // or Impl* pImpl;
};

// *.cpp
#include "typedb/common/native.hpp"

struct DatabaseManager::Impl {
    TypeDBNative::DatabaseManager* native;
    ~Impl() { database_manager_drop(native); } // if using `unique_ptr<Impl>`
}
```
More info: https://en.cppreference.com/w/cpp/language/pimpl

Reply:
They're being leaked, but not into the TypeDB namespace. Just into the TypeDBNative namespace.

This does make sense, but I guess the pImpl will always be on the heap then? Would we still want to do it for classes that we're instantiating plenty of times (e.g. Concept)?

Reply:
After a coffee table conversation, we decided we'd avoid PIMPL, keep things simple and import the native symbols into the nested namespace `TypeDB::_native` instead.